### ISSUE TITLE
Doesn't work on Windows

### ISSUE THREAD

**Issue by CryZe** [vickenty/lang-c#1]
It seems a lot of assumptions about the preprocessor are made that don't apply to the preprocessor on Windows.

**Comment by vickenty** [comment]
Indeed, lang-c currently relies on gcc to do preprocessing. I don't know if msvc command-line compiler can be used as a stand-alone preprocessor, but it would be nice to support it if it is. PRs welcome :)

**Comment by Artoria2e5** [comment]
There seems to be [`cl /p`](https://learn.microsoft.com/en-us/cpp/build/reference/p-preprocess-to-a-file?view=msvc-170)
